packages/emitter-framework/src/python/components/init-declaration.tsx

```diff
@@ -0,0 +1,53 @@
+import { children, Children, Declaration, DeclarationProps, mapJoin, Scope } from "@alloy-js/core";
+import { ParameterDescriptor } from "./parameter.jsx";
+import { usePythonNamePolicy } from "../name-policy.js";
+
+export interface InitDeclarationProps extends Omit<DeclarationProps, "name"> {
+  parameters?: Record<string, Children | ParameterDescriptor>;
+}
+
+function isParameterDescriptor(value: Children | ParameterDescriptor): value is ParameterDescriptor {
+  return (
+    typeof value === "object" && value !== null && Object.hasOwn(value, "type")
+  );
+}
+
+export function InitDeclaration(props: InitDeclarationProps) {
+  const name = "__init__";
+  const args = {self: "", ...props.parameters ?? {}};
+  const argsClause = mapJoin(Object.entries(args), ([name, param]) => {
+    const pythonName = usePythonNamePolicy().getName(name, "parameter");
+    if (isParameterDescriptor(param)) {
+      return `${pythonName}: ${param.type}`;
+    } else if (param && param !== "") {
+      return `${pythonName}: ${param}`
+    } else {
+      return pythonName;
+    }
+  },
+  {joiner: ", "});
+  let childrenClause: Children;
+  if (!props.children && !props.parameters) {
+    childrenClause = "pass";
+  } else if (!props.children && props.parameters) {
+    // map parameters to self assignments
+    childrenClause = mapJoin(Object.entries(props.parameters), ([paramName, param]) => {
+      const pythonName = usePythonNamePolicy().getName(paramName, "parameter");
+      if (isParameterDescriptor(param)) {
+        return `self.${pythonName} = ${pythonName}`;
+      } else {
+        return `self.${pythonName} = ${param}`;
+      }
+    }, {joiner: "\n"});
+  } else {
+    childrenClause = props.children;
+  }
+  return (
+    <Declaration {...props} name={name}>
```

Comment:
This is from my lack of Python knowledge, can you reference __init__ in other places?

Reply:
You can do things like
```Python
foo = Foo.__init__
```